File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message (logLine: string): string =
    (logLine.Trim().Remove(0,3+logLine.Trim().IndexOf("]:"))).Trim()
let logLevel(logLine: string): string = 
    logLine.Remove(logLine.IndexOf("]:")).Remove(0,1).ToLower()
let reformat(logLine: string): string = 
    message(logLine)+" ("+logLevel(logLine)+")"

File: solutions/csharp/log-levels/1/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        return (logLine.Substring(logLine.IndexOf("]:")+3, logLine.Length - logLine.IndexOf("]:") - 3)).Trim();
    }

    public static string LogLevel(string logLine)
    {
        return (logLine.Substring(logLine.IndexOf("[")+1, logLine.IndexOf("]")-1)).ToLower().Trim();
    }

    public static string Reformat(string logLine)
    {
        return Message(logLine)+" ("+LogLevel(logLine)+")";
    }
}
